brave/brave-core
# OS(MacOS/Windows) theme controls brave theme

Please see each commit's log.
Fix https://github.com/brave/brave-browser/issues/1189
Fix https://github.com/brave/brave-browser/issues/1289

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
`yarn test brave_browser_tests --filter=BraveThemeService*`
`yarn test brave_unit_tests --filter=BraveThemeServiceTest*`

Manual test step
1. Launch Browser
2. Check theme settings have three options (`Same as MacOS`, `Dark`, `Light`)
3. Check `Same as MacOS` applies MacOS theme to brave theme
4. Check Dark/Light works independently with MacOS theme.

## Reviewer Checklist:
